src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -316,19 +224,171 @@ public String constructCommandForTest(String inputArguments) {
+
+        @DataBoundConstructor
+        public PdfArtifact(String pdfReportFilePath) {
+            super(pdfReportFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(PDF_REPORT_PATH, getFilePath());
+        }
+    }
+
+    public static class TapArtifact extends AbstractArtifactImpl {
+
+        private static final String TAP_RESULTS_PATH = "TAPResultsPath";
+
+        @DataBoundConstructor
+        public TapArtifact(String tapReportFilePath) {
+            super(tapReportFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(TAP_RESULTS_PATH, getFilePath());
+        }
+    }
+
+    public static class JunitArtifact extends AbstractArtifactImpl {
+
+        private static final String JUNIT_RESULTS_PATH = "JUnitResultsPath";
+
+        @DataBoundConstructor
+        public JunitArtifact(String junitReportFilePath) {
+            super(junitReportFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(JUNIT_RESULTS_PATH, getFilePath());
+        }
+    }
+
+    public static class CoberturaArtifact extends AbstractArtifactImpl {
+
+        private static final String COBERTURA_CODE_COVERAGE_PATH = "CoberturaCodeCoveragePath";
+
+        @DataBoundConstructor
+        public CoberturaArtifact(String coberturaReportFilePath) {
+            super(coberturaReportFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(COBERTURA_CODE_COVERAGE_PATH, getFilePath());
+        }
+    }
+
+    public static class StmResultsArtifact extends AbstractArtifactImpl {
+
+        private static final String STM_RESULTS_PATH = "SimulinkTestResultsPath";
+
+        @DataBoundConstructor
+        public StmResultsArtifact(String stmResultsFilePath) {
+            super(stmResultsFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(STM_RESULTS_PATH, getFilePath());
+        }
+    }
+
+    public static class ModelCovArtifact extends AbstractArtifactImpl {
+
+        private static final String COBERTURA_MODEL_COVERAGE_PATH = "CoberturaModelCoveragePath";
+
+        @DataBoundConstructor
+        public ModelCovArtifact(String modelCoverageFilePath) {
+            super(modelCoverageFilePath);
+        }
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(COBERTURA_MODEL_COVERAGE_PATH, getFilePath());
+        }
+    }
+
+    public static class NullArtifact implements Artifact {
+
+        @Override
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+
+        }
+
+        @Override
+        public boolean getDefault() {

```

Comment:
Since this is returning false for the unchecked case, you really should rename this to be getSelected or getIsSelected, and the resulting jelly file of course would need updating

Reply:
(or name it something like that. getDefault is wrong now because default applies to when the artifact is not selected so should return true for that naming)

Reply:
Done.